hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/LocalMetadataStore.java

```diff
@@ -192,15 +202,19 @@ public synchronized DirListingMetadata listChildren(Path p) throws
   }
 
   @Override
-  public void move(Collection<Path> pathsToDelete,
-      Collection<PathMetadata> pathsToCreate,
-      ITtlTimeProvider ttlTimeProvider) throws IOException {
+  public void move(
+      @Nullable Collection<Path> pathsToDelete,
+      @Nullable Collection<PathMetadata> pathsToCreate,
+        ITtlTimeProvider ttlTimeProvider,
```

Comment:
I still don't immediately seeing ttlTimeProvider being used everywhere it's passed. Did we get to the bottom of that? Now's the time to remove it, maybe.

Reply:
Yes we did. I started a discussion about this if we want to pass it in the metastore init instead of every method which will use it. We ended up passing it to every method instead of the init method. We need to fix that. If that is not fixed in this pr I will fix it tomorrow under a new issue.

Reply:
Makes sense. If you need to override for a single test operation, can always add a @VisibleForTesting setter.